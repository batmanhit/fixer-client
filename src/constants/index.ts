export const bankAccountFormStepFields = [
  ['bankId', 'branchId'],
  ['accountHolderName', 'accountType', 'accountNumber'],
  ['employeeName', 'employeeNumber'],
];

export const bankAccountFormValidationRules = {
  bankId: 'required',
  branchId: 'required',
  accountHolderName: 'required',
  accountType: 'required|in:Savings,Checking',
  accountNumber: 'required|digits:7',
  employeeName: 'required',
  employeeNumber: 'required|digits:15',
};

export const customBankAccountFormValidationNames = {
  'bankId': 'Bank name',
  'branchId': 'Branch name',
  'accountHolderName': 'Account holder\'s name',
  'accountType': 'Account type',
  'accountNumber': 'Account number',
  'employeeName': 'Employee name',
  'employeeNumber': 'Employee number',
}