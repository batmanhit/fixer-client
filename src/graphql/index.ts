export * from './query/bankAccount';
export * from './query/bankAccounts';
export * from './query/banks';
export * from './query/branches';

export * from './mutation/createBankAccount';
export * from './mutation/deleteBankAccount';
export * from './mutation/deleteBankAccounts';
export * from './mutation/updateBankAccount';