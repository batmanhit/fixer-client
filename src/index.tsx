import * as React from 'react';

import ApolloClient from "apollo-boost";
import 'bootstrap/dist/css/bootstrap.css';
import { ApolloProvider } from "react-apollo";
import * as ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';

import App from './containers/app';

import './index.css';
import registerServiceWorker from './registerServiceWorker';

const client = new ApolloClient({
  uri: "http://localhost:8000/graphql"
});

ReactDOM.render(
  <Router>
    <ApolloProvider client={client}>
      <App />
    </ApolloProvider>
  </Router>,
  document.getElementById('root') as HTMLElement
);
registerServiceWorker();
