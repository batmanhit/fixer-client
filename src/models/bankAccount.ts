import { Bank } from './bank';
import { Branch } from './branch';

export interface BankAccount {
  id: string;
  accountHolderName: string;
  employeeName: string;
  accountType: string;
  accountNumber: string;
  employeeNumber: string;
  bank: Bank;
  branch: Branch;
  createdAt: Date;
  updatedAt: Date;
};

export interface BankAccountInfo {
  accountHolderName: string;
  employeeName: string;
  accountType: string;
  accountNumber: string;
  employeeNumber: string;
  bankId: string;
  branchId: string;
};

export const convertFromBankAccountToInfo = (bankAccount: BankAccount): BankAccountInfo => {
  return {
    accountHolderName: bankAccount.accountHolderName,
    employeeName: bankAccount.employeeName,
    accountType: bankAccount.accountType,
    accountNumber: bankAccount.accountNumber,
    employeeNumber: bankAccount.employeeNumber,
    bankId: bankAccount.bank.id,
    branchId: bankAccount.branch.id
  }
}