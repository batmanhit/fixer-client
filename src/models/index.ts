export * from './bankAccount';
export * from './bank';
export * from './branch';
