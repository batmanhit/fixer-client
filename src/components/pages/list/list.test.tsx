import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { AgGridReact } from 'ag-grid-react/lib/agGridReact';

import { BankAccount } from '../../../../src/models';

import List from './list';

describe('BankAccountsList', () => {
  const mockBankAccounts: Array<BankAccount> = [
    {
      id: '1',
      accountHolderName: 'Test1',
      employeeName: 'Employee1',
      accountType: 'Savings',
      accountNumber: '1234567',
      employeeNumber: '123456789012345',
      bank: {
        id: '1',
        name: 'Bank1'
      },
      branch: {
        id: '1',
        name: 'Branch1'
      },
      createdAt: new Date('2019/04/21 00:00:00'),
      updatedAt: new Date('2019/04/21 00:00:00')
    }, {
      id: '2',
      accountHolderName: 'Test2',
      employeeName: 'Employee2',
      accountType: 'Checking',
      accountNumber: '1234567',
      employeeNumber: '123456789012345',
      bank: {
        id: '2',
        name: 'Bank2'
      },
      branch: {
        id: '2',
        name: 'Branch2'
      },
      createdAt: new Date('2019/04/21 00:00:00'),
      updatedAt: new Date('2019/04/21 00:00:00')
    },
  ];
  const updateItem = jest.fn();
  const deleteItem = jest.fn();
  const testElement: JSX.Element = <List
    data={mockBankAccounts}
    onUpdateItem={updateItem}
    onDeleteItem={deleteItem}
  />;
  const wrapper: ShallowWrapper = shallow(testElement);

  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('render mockBankAccounts correctly', () => {
    const grid = wrapper.find(AgGridReact);
    expect(grid.prop('rowData')).toEqual(mockBankAccounts);
  });
});