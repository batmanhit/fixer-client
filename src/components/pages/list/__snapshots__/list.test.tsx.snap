// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankAccountsList snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    data={
        Array [
            Object {
              "accountHolderName": "Test1",
              "accountNumber": "1234567",
              "accountType": "Savings",
              "bank": Object {
                "id": "1",
                "name": "Bank1",
              },
              "branch": Object {
                "id": "1",
                "name": "Branch1",
              },
              "createdAt": 2019-04-20T16:00:00.000Z,
              "employeeName": "Employee1",
              "employeeNumber": "123456789012345",
              "id": "1",
              "updatedAt": 2019-04-20T16:00:00.000Z,
            },
            Object {
              "accountHolderName": "Test2",
              "accountNumber": "1234567",
              "accountType": "Checking",
              "bank": Object {
                "id": "2",
                "name": "Bank2",
              },
              "branch": Object {
                "id": "2",
                "name": "Branch2",
              },
              "createdAt": 2019-04-20T16:00:00.000Z,
              "employeeName": "Employee2",
              "employeeNumber": "123456789012345",
              "id": "2",
              "updatedAt": 2019-04-20T16:00:00.000Z,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="ag-theme-bootstrap"
        style={
                Object {
                        "height": "calc(100% - 100px)",
                        "width": "100%",
                      }
        }
>
        <AgGridReact
                columnDefs={
                        Array [
                                Object {
                                  "field": "accountHolderName",
                                  "headerName": "Account Holder Name",
                                },
                                Object {
                                  "field": "accountType",
                                  "headerName": "Account Type",
                                },
                                Object {
                                  "field": "accountNumber",
                                  "filter": "agNumberColumnFilter",
                                  "headerName": "Account Number",
                                },
                                Object {
                                  "field": "employeeName",
                                  "headerName": "Employee Name",
                                },
                                Object {
                                  "field": "employeeNumber",
                                  "filter": "agNumberColumnFilter",
                                  "headerName": "Employee Number",
                                },
                                Object {
                                  "field": "updatedAt",
                                  "filter": "agDateColumnFilter",
                                  "headerName": "Updated At",
                                },
                              ]
                }
                defaultColDef={
                        Object {
                                "filter": true,
                                "sortable": true,
                              }
                }
                rowData={
                        Array [
                                Object {
                                  "accountHolderName": "Test1",
                                  "accountNumber": "1234567",
                                  "accountType": "Savings",
                                  "bank": Object {
                                    "id": "1",
                                    "name": "Bank1",
                                  },
                                  "branch": Object {
                                    "id": "1",
                                    "name": "Branch1",
                                  },
                                  "createdAt": 2019-04-20T16:00:00.000Z,
                                  "employeeName": "Employee1",
                                  "employeeNumber": "123456789012345",
                                  "id": "1",
                                  "updatedAt": 2019-04-20T16:00:00.000Z,
                                },
                                Object {
                                  "accountHolderName": "Test2",
                                  "accountNumber": "1234567",
                                  "accountType": "Checking",
                                  "bank": Object {
                                    "id": "2",
                                    "name": "Bank2",
                                  },
                                  "branch": Object {
                                    "id": "2",
                                    "name": "Branch2",
                                  },
                                  "createdAt": 2019-04-20T16:00:00.000Z,
                                  "employeeName": "Employee2",
                                  "employeeNumber": "123456789012345",
                                  "id": "2",
                                  "updatedAt": 2019-04-20T16:00:00.000Z,
                                },
                              ]
                }
        />
</div>,
      "className": "h-100 p-3",
      "fluid": true,
      "tag": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <AgGridReact
          columnDefs={
                    Array [
                              Object {
                                "field": "accountHolderName",
                                "headerName": "Account Holder Name",
                              },
                              Object {
                                "field": "accountType",
                                "headerName": "Account Type",
                              },
                              Object {
                                "field": "accountNumber",
                                "filter": "agNumberColumnFilter",
                                "headerName": "Account Number",
                              },
                              Object {
                                "field": "employeeName",
                                "headerName": "Employee Name",
                              },
                              Object {
                                "field": "employeeNumber",
                                "filter": "agNumberColumnFilter",
                                "headerName": "Employee Number",
                              },
                              Object {
                                "field": "updatedAt",
                                "filter": "agDateColumnFilter",
                                "headerName": "Updated At",
                              },
                            ]
          }
          defaultColDef={
                    Object {
                              "filter": true,
                              "sortable": true,
                            }
          }
          rowData={
                    Array [
                              Object {
                                "accountHolderName": "Test1",
                                "accountNumber": "1234567",
                                "accountType": "Savings",
                                "bank": Object {
                                  "id": "1",
                                  "name": "Bank1",
                                },
                                "branch": Object {
                                  "id": "1",
                                  "name": "Branch1",
                                },
                                "createdAt": 2019-04-20T16:00:00.000Z,
                                "employeeName": "Employee1",
                                "employeeNumber": "123456789012345",
                                "id": "1",
                                "updatedAt": 2019-04-20T16:00:00.000Z,
                              },
                              Object {
                                "accountHolderName": "Test2",
                                "accountNumber": "1234567",
                                "accountType": "Checking",
                                "bank": Object {
                                  "id": "2",
                                  "name": "Bank2",
                                },
                                "branch": Object {
                                  "id": "2",
                                  "name": "Branch2",
                                },
                                "createdAt": 2019-04-20T16:00:00.000Z,
                                "employeeName": "Employee2",
                                "employeeNumber": "123456789012345",
                                "id": "2",
                                "updatedAt": 2019-04-20T16:00:00.000Z,
                              },
                            ]
          }
/>,
        "className": "ag-theme-bootstrap",
        "style": Object {
          "height": "calc(100% - 100px)",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "columnDefs": Array [
            Object {
              "field": "accountHolderName",
              "headerName": "Account Holder Name",
            },
            Object {
              "field": "accountType",
              "headerName": "Account Type",
            },
            Object {
              "field": "accountNumber",
              "filter": "agNumberColumnFilter",
              "headerName": "Account Number",
            },
            Object {
              "field": "employeeName",
              "headerName": "Employee Name",
            },
            Object {
              "field": "employeeNumber",
              "filter": "agNumberColumnFilter",
              "headerName": "Employee Number",
            },
            Object {
              "field": "updatedAt",
              "filter": "agDateColumnFilter",
              "headerName": "Updated At",
            },
          ],
          "defaultColDef": Object {
            "filter": true,
            "sortable": true,
          },
          "rowData": Array [
            Object {
              "accountHolderName": "Test1",
              "accountNumber": "1234567",
              "accountType": "Savings",
              "bank": Object {
                "id": "1",
                "name": "Bank1",
              },
              "branch": Object {
                "id": "1",
                "name": "Branch1",
              },
              "createdAt": 2019-04-20T16:00:00.000Z,
              "employeeName": "Employee1",
              "employeeNumber": "123456789012345",
              "id": "1",
              "updatedAt": 2019-04-20T16:00:00.000Z,
            },
            Object {
              "accountHolderName": "Test2",
              "accountNumber": "1234567",
              "accountType": "Checking",
              "bank": Object {
                "id": "2",
                "name": "Bank2",
              },
              "branch": Object {
                "id": "2",
                "name": "Branch2",
              },
              "createdAt": 2019-04-20T16:00:00.000Z,
              "employeeName": "Employee2",
              "employeeNumber": "123456789012345",
              "id": "2",
              "updatedAt": 2019-04-20T16:00:00.000Z,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="ag-theme-bootstrap"
          style={
                    Object {
                              "height": "calc(100% - 100px)",
                              "width": "100%",
                            }
          }
>
          <AgGridReact
                    columnDefs={
                              Array [
                                        Object {
                                          "field": "accountHolderName",
                                          "headerName": "Account Holder Name",
                                        },
                                        Object {
                                          "field": "accountType",
                                          "headerName": "Account Type",
                                        },
                                        Object {
                                          "field": "accountNumber",
                                          "filter": "agNumberColumnFilter",
                                          "headerName": "Account Number",
                                        },
                                        Object {
                                          "field": "employeeName",
                                          "headerName": "Employee Name",
                                        },
                                        Object {
                                          "field": "employeeNumber",
                                          "filter": "agNumberColumnFilter",
                                          "headerName": "Employee Number",
                                        },
                                        Object {
                                          "field": "updatedAt",
                                          "filter": "agDateColumnFilter",
                                          "headerName": "Updated At",
                                        },
                                      ]
                    }
                    defaultColDef={
                              Object {
                                        "filter": true,
                                        "sortable": true,
                                      }
                    }
                    rowData={
                              Array [
                                        Object {
                                          "accountHolderName": "Test1",
                                          "accountNumber": "1234567",
                                          "accountType": "Savings",
                                          "bank": Object {
                                            "id": "1",
                                            "name": "Bank1",
                                          },
                                          "branch": Object {
                                            "id": "1",
                                            "name": "Branch1",
                                          },
                                          "createdAt": 2019-04-20T16:00:00.000Z,
                                          "employeeName": "Employee1",
                                          "employeeNumber": "123456789012345",
                                          "id": "1",
                                          "updatedAt": 2019-04-20T16:00:00.000Z,
                                        },
                                        Object {
                                          "accountHolderName": "Test2",
                                          "accountNumber": "1234567",
                                          "accountType": "Checking",
                                          "bank": Object {
                                            "id": "2",
                                            "name": "Bank2",
                                          },
                                          "branch": Object {
                                            "id": "2",
                                            "name": "Branch2",
                                          },
                                          "createdAt": 2019-04-20T16:00:00.000Z,
                                          "employeeName": "Employee2",
                                          "employeeNumber": "123456789012345",
                                          "id": "2",
                                          "updatedAt": 2019-04-20T16:00:00.000Z,
                                        },
                                      ]
                    }
          />
</div>,
        "className": "h-100 p-3",
        "fluid": true,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <AgGridReact
            columnDefs={
                        Array [
                                    Object {
                                      "field": "accountHolderName",
                                      "headerName": "Account Holder Name",
                                    },
                                    Object {
                                      "field": "accountType",
                                      "headerName": "Account Type",
                                    },
                                    Object {
                                      "field": "accountNumber",
                                      "filter": "agNumberColumnFilter",
                                      "headerName": "Account Number",
                                    },
                                    Object {
                                      "field": "employeeName",
                                      "headerName": "Employee Name",
                                    },
                                    Object {
                                      "field": "employeeNumber",
                                      "filter": "agNumberColumnFilter",
                                      "headerName": "Employee Number",
                                    },
                                    Object {
                                      "field": "updatedAt",
                                      "filter": "agDateColumnFilter",
                                      "headerName": "Updated At",
                                    },
                                  ]
            }
            defaultColDef={
                        Object {
                                    "filter": true,
                                    "sortable": true,
                                  }
            }
            rowData={
                        Array [
                                    Object {
                                      "accountHolderName": "Test1",
                                      "accountNumber": "1234567",
                                      "accountType": "Savings",
                                      "bank": Object {
                                        "id": "1",
                                        "name": "Bank1",
                                      },
                                      "branch": Object {
                                        "id": "1",
                                        "name": "Branch1",
                                      },
                                      "createdAt": 2019-04-20T16:00:00.000Z,
                                      "employeeName": "Employee1",
                                      "employeeNumber": "123456789012345",
                                      "id": "1",
                                      "updatedAt": 2019-04-20T16:00:00.000Z,
                                    },
                                    Object {
                                      "accountHolderName": "Test2",
                                      "accountNumber": "1234567",
                                      "accountType": "Checking",
                                      "bank": Object {
                                        "id": "2",
                                        "name": "Bank2",
                                      },
                                      "branch": Object {
                                        "id": "2",
                                        "name": "Branch2",
                                      },
                                      "createdAt": 2019-04-20T16:00:00.000Z,
                                      "employeeName": "Employee2",
                                      "employeeNumber": "123456789012345",
                                      "id": "2",
                                      "updatedAt": 2019-04-20T16:00:00.000Z,
                                    },
                                  ]
            }
/>,
          "className": "ag-theme-bootstrap",
          "style": Object {
            "height": "calc(100% - 100px)",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "columnDefs": Array [
              Object {
                "field": "accountHolderName",
                "headerName": "Account Holder Name",
              },
              Object {
                "field": "accountType",
                "headerName": "Account Type",
              },
              Object {
                "field": "accountNumber",
                "filter": "agNumberColumnFilter",
                "headerName": "Account Number",
              },
              Object {
                "field": "employeeName",
                "headerName": "Employee Name",
              },
              Object {
                "field": "employeeNumber",
                "filter": "agNumberColumnFilter",
                "headerName": "Employee Number",
              },
              Object {
                "field": "updatedAt",
                "filter": "agDateColumnFilter",
                "headerName": "Updated At",
              },
            ],
            "defaultColDef": Object {
              "filter": true,
              "sortable": true,
            },
            "rowData": Array [
              Object {
                "accountHolderName": "Test1",
                "accountNumber": "1234567",
                "accountType": "Savings",
                "bank": Object {
                  "id": "1",
                  "name": "Bank1",
                },
                "branch": Object {
                  "id": "1",
                  "name": "Branch1",
                },
                "createdAt": 2019-04-20T16:00:00.000Z,
                "employeeName": "Employee1",
                "employeeNumber": "123456789012345",
                "id": "1",
                "updatedAt": 2019-04-20T16:00:00.000Z,
              },
              Object {
                "accountHolderName": "Test2",
                "accountNumber": "1234567",
                "accountType": "Checking",
                "bank": Object {
                  "id": "2",
                  "name": "Bank2",
                },
                "branch": Object {
                  "id": "2",
                  "name": "Branch2",
                },
                "createdAt": 2019-04-20T16:00:00.000Z,
                "employeeName": "Employee2",
                "employeeNumber": "123456789012345",
                "id": "2",
                "updatedAt": 2019-04-20T16:00:00.000Z,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
